File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    let answer = [];
    
    commands.forEach((item) => {
        let i = item[0]-1;
        let j = item[1];
        let k = item[2]-1;
        answer.push(array.slice(i, j).sort((a,b) => a-b)[k]);
    })
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        // commands 배열의 길이만큼 반복
        for(int i=0; i<commands.length; i++){
            // 새로운 자른 배열 만들기
            int[] cutArr = Arrays.copyOfRange(array, (commands[i][0]-1), (commands[i][1]));
            
            // 정렬(오름차순)
            Arrays.sort(cutArr);
            
            // n번째 숫자 가져오기
            answer[i] = cutArr[commands[i][2]-1];
        }
        
        return answer;
    }
}